Neos.Flow/Classes/Mvc/ActionRequest.php

```diff
@@ -695,6 +697,17 @@ public function getFormat(): string
         return $this->format;
     }
 
+    /**
+     * Returns the base Uri of the request.
+     *
+     * @return UriInterface The base URI of the request
+     * @api
+     */
+    public function getBaseUri(): UriInterface
+    {
+        return $this->getHttpRequest()->getAttribute(ServerRequestAttributes::BASE_URI);
```

Comment:
That still means we rely on that attribute. I would very much like to get rid of it at some point.

Reply:
Generally yes, but that's not easy, as there are currently a couple of places that only have access to the Http Request and still use the baseUri, e.g. the WebRedirect or FileSystemTarget

Reply:
Sure, I think we should discuss this in more detail with basti as well, IMHO we should figure out the intend in various places and adjust accordingly. Eg. FileSystemTarget is weird anyway. And components could decide between setting and building a baseUri from the request on their own (or with an additional indpeendent helper/service)

Reply:
Like it's described right now I'd expect this method to return the base URI of the current HTTP request. But it might not if that is different from the configured `Neos.Flow.http.baseUri`.

Reply:
I'd expect that probably in the Http Request, but in the ActionRequest, I expect the baseUri that would be used for any rendering of a link for example (which is either the baseUri of the request, or can be overridden by the configuration), so that's exactly what it does. But sure, let's have a chat about that.